packages/math/src/PoolMath/StableSwapsPool.ts

```diff
@@ -0,0 +1,401 @@
+import { SharedPoolMath } from "./index.js";
+import { Fraction, type TFractionLike } from "@sundaeswap/fraction";
+
+export function liquidityInvariant(
+  assetA: bigint,
+  assetB: bigint,
+  linearAmplification: bigint,
+  newSumInvariant: bigint,
+) {
+  // 4 * A * 4 * (x*y) * D + D^3 - (4(x*y) * (4A(x + y) + D))
+  // 4A * 4xy * D + D^3 - (4xy * 4a * (x + y) + 4xy * D)
+  // 16Axy * D + D^3 - (16Axy * (x + y) + 4xy * D)
+  const four_a = 4n * linearAmplification;
+  const four_x_y = 4n * assetA * assetB;
+  const d_plus_one = newSumInvariant + 1n;
+  const d_cubed = newSumInvariant * newSumInvariant * newSumInvariant;
+  const d_plus_one_cubed = d_plus_one * d_plus_one * d_plus_one;
+  const x_plus_y = assetA + assetB;
+  const sixteen_a_x_y = four_a * four_x_y;
+  const sixteen_a_x_y_x_plus_y = sixteen_a_x_y * x_plus_y;
+  const f1 =
+    sixteen_a_x_y * newSumInvariant +
+    d_cubed -
+    (sixteen_a_x_y_x_plus_y + four_x_y * newSumInvariant);
+  const f2 =
+    sixteen_a_x_y * d_plus_one +
+    d_plus_one_cubed -
+    (sixteen_a_x_y_x_plus_y + four_x_y * d_plus_one);
+  return f1 <= 0n && f2 > 0n;
+}
+
+export const aPrecision = 200n;
```

Comment:
Typical convention we've used for exporting static constants is to make them all caps. i.e. `A_PRECISION`